src/model/model.nit

```diff
@@ -2602,3 +2615,70 @@ fun interface_kind: MClassKind do return once new MClassKind("interface", false)
 fun enum_kind: MClassKind do return once new MClassKind("enum", false)
 # The class kind `extern`
 fun extern_kind: MClassKind do return once new MClassKind("extern class", false)
+
+# A standalone pre-constructed model used to test various model-related methods.
+#
+# When instantiated, a standalone model is already filled with entities that are exposed as attributes.
+class ModelStandalone
+	super Model
+
+	redef var location = new Location.opaque_file("ModelStandalone")
+
+	# The first module
+	var mmodule0 = new MModule(self, null, "module0", location)
+
+	# The root Object class
+	var mclass_o = new MClass(mmodule0, "Object", location, null, interface_kind, public_visibility)
+
+	# The introduction of `mclass_o`
+	var mclassdef_o = new MClassDef(mmodule0, mclass_o.mclass_type, location)
+end
+
+# A standalone model with the common class diamond-hierarchy ABCD
+class ModelDiamond
```

Comment:
Nitpicking: It should be `DiamondModel` as the adjective normally comes before the noun.